vm/src/value.rs

```diff
@@ -245,6 +245,18 @@ impl Value {
         matches!(&self, Value::Byte(_) | Value::Int(_) | Value::Float(_))
     }
 
+    pub fn is_float(&self) -> bool {
+        matches!(self, Value::Float(_))
+    }
+
+    pub fn not(&self) -> Value {
+        match self {
+            Value::False => Value::True,
```

Comment:
Change this to
```rust
Value::False | Value::Nil => Value::True
```
match statements in rust are super flexible. It's really cool.

On some timeline this can be a very cool read:
https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html

because there's a lot you can do.
It's also worth noting that the matches! macro essentially wraps an actual match statement so you can use all this funky syntax in there too.